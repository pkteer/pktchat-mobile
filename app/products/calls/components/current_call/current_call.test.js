// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {TouchableOpacity} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';

import CurrentCall from './current_call';

describe('CurrentCall', () => {
    const baseProps = {
        actions: {
            muteMyself: jest.fn(),
            unmuteMyself: jest.fn(),
        },
        theme: Preferences.THEMES.denim,
        channel: {
            display_name: 'Channel Name',
        },
        speaker: {
            id: 'user-1-id',
            muted: false,
            isTalking: true,
        },
        speakerUser: {
            id: 'user-1-id',
            username: 'user-1-username',
            nickname: 'User 1',
        },
        call: {
            participants: {
                'user-1-id': {
                    id: 'user-1-id',
                    muted: false,
                    isTalking: false,
                },
                'user-2-id': {
                    id: 'user-2-id',
                    muted: true,
                    isTalking: true,
                },
            },
            channelId: 'channel-id',
            startTime: 100,
            speakers: 'user-2-id',
            screenOn: false,
            threadId: false,
        },
        currentParticipant: {
            id: 'user-2-id',
            muted: true,
            isTalking: true,
        },
        teammateNameDisplay: Preferences.DISPLAY_PREFER_NICKNAME,
    };

    test('should match snapshot muted', () => {
        const props = {...baseProps, currentParticipant: {...baseProps.currentParticipant, muted: true}};
        const wrapper = shallow(<CurrentCall {...props}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot unmuted', () => {
        const props = {...baseProps, currentParticipant: {...baseProps.currentParticipant, muted: false}};
        const wrapper = shallow(<CurrentCall {...props}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should mute on click mute button', () => {
        const muteMyself = jest.fn();
        const unmuteMyself = jest.fn();
        const props = {...baseProps, actions: {muteMyself, unmuteMyself}, currentParticipant: {...baseProps.currentParticipant, muted: false}};
        const wrapper = shallow(<CurrentCall {...props}/>);

        wrapper.find(TouchableOpacity).simulate('press');
        expect(props.actions.muteMyself).toHaveBeenCalled();
        expect(props.actions.unmuteMyself).not.toHaveBeenCalled();
    });

    test('should ask for confirmation on click', () => {
        const muteMyself = jest.fn();
        const unmuteMyself = jest.fn();
        const props = {...baseProps, actions: {unmuteMyself, muteMyself}, currentParticipant: {...baseProps.currentParticipant, muted: true}};
        const wrapper = shallow(<CurrentCall {...props}/>);

        wrapper.find(TouchableOpacity).simulate('press');
        expect(props.actions.muteMyself).not.toHaveBeenCalled();
        expect(props.actions.unmuteMyself).toHaveBeenCalled();
    });
});
