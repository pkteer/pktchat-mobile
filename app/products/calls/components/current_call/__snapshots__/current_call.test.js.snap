// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentCall should match snapshot muted 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3F4350",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 64,
        "padding": 4,
        "width": "100%",
      }
    }
  >
    <CallAvatar
      userId=""
      volume={0}
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        No one is talking
      </Text>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "opacity": 0.64,
          }
        }
      >
        ~Channel Name
      </Text>
    </View>
    <Pressable
      onPressIn={[Function]}
      style={
        Object {
          "zIndex": 10,
        }
      }
    >
      <CompassIcon
        name="arrow-expand"
        size={24}
        style={
          Object {
            "color": "#ffffff",
            "marginRight": 8,
            "padding": 8,
          }
        }
      />
    </Pressable>
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "zIndex": 10,
        }
      }
    >
      <CompassIcon
        name="microphone-off"
        size={24}
        style={
          Array [
            Object {
              "backgroundColor": "#3DB887",
              "borderRadius": 4,
              "color": "#ffffff",
              "height": 42,
              "justifyContent": "center",
              "margin": 4,
              "overflow": "hidden",
              "padding": 9,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 42,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;

exports[`CurrentCall should match snapshot unmuted 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3F4350",
        "borderRadius": 5,
        "flexDirection": "row",
        "height": 64,
        "padding": 4,
        "width": "100%",
      }
    }
  >
    <CallAvatar
      userId=""
      volume={0}
    />
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        No one is talking
      </Text>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "opacity": 0.64,
          }
        }
      >
        ~Channel Name
      </Text>
    </View>
    <Pressable
      onPressIn={[Function]}
      style={
        Object {
          "zIndex": 10,
        }
      }
    >
      <CompassIcon
        name="arrow-expand"
        size={24}
        style={
          Object {
            "color": "#ffffff",
            "marginRight": 8,
            "padding": 8,
          }
        }
      />
    </Pressable>
    <TouchableOpacity
      onPress={[Function]}
      style={
        Object {
          "zIndex": 10,
        }
      }
    >
      <CompassIcon
        name="microphone"
        size={24}
        style={
          Array [
            Object {
              "backgroundColor": "#3DB887",
              "borderRadius": 4,
              "color": "#ffffff",
              "height": 42,
              "justifyContent": "center",
              "margin": 4,
              "overflow": "hidden",
              "padding": 9,
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 42,
            },
            undefined,
          ]
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
